nemo/collections/tts/modules/magpietts_inference/evaluate_generated_audio.py

```diff
@@ -318,19 +326,29 @@ def evaluate(
                 sv_model_type=sv_model_type,
             )
 
-            # Ground truth vs. predicted
-            gt_speaker_embedding = extract_embedding_fn(audio_path=gt_audio_filepath)
-            pred_speaker_embedding = extract_embedding_fn(audio_path=pred_audio_filepath)
-            pred_gt_ssim = torch.nn.functional.cosine_similarity(
-                gt_speaker_embedding, pred_speaker_embedding, dim=0
-            ).item()
+            # Initialize SSIMs with a default since the contest or ground truth audio
+            # may be unavailable.
+            pred_context_ssim = 0.0
+            gt_context_ssim = 0.0
+            pred_context_ssim_alternate = 0.0
+            gt_context_ssim_alternate = 0.0
+            pred_gt_ssim = 0.0
+            pred_gt_ssim_alternate = 0.0
```

Comment:
While this seems ok, wouldn't it be better to default to None or NaN instead? Or will that break some averaging logic down the road?
Might be a better idea to stick to None or NaN and check that the dataset of interest does not contain context audios.